Jenkinsfile

```diff
@@ -1,97 +1,67 @@
-          }
-        }
-      }
-    }
-
-    stage('Initialize Job Details') {
       steps {
-        script {
-          currentBuild.description = "${env.SERVICE} : ${env.GIT_COMMIT[0..5]} : ${env.NODE_NAME}"
-        }
+        sh 'sbt assembly'
       }
     }
-
-    stage('Master') {
+    stage('Build') {
       when { branch 'main' }
       stages {
         stage('Build and push image') {
           steps {
-            sh 'sbt assembly'
+            script {
+              lastStage = env.STAGE_NAME
+              sh 'sbt assembly'
 
-            // NOTE: If you're changing the version of this service, you'll also have to change the version here
-            sh 'cp target/scala-2.12/image-resize-service-assembly-0.0.1.jar image-resize-service-assembly.jar'
-            sh 'docker build --tag image-resize-service:latest .'
-            sh 'rm image-resize-service-assembly.jar'
+              // NOTE: If you're changing the version of this service, you'll also have to change the version here
+              sh 'cp target/scala-2.12/image-resize-service-assembly-0.0.1.jar image-resize-service-assembly.jar'
+              sh 'docker build --tag image-resize-service:latest .'
+              sh 'rm image-resize-service-assembly.jar'
 
-            script {
               env.DOCKER_TAG = dockerize.push_tagged_image_to_all_repos('image-resize-service:latest', env.SERVICE_VERSION, env.GIT_COMMIT)
-            }
-          }
-          post {
-            failure {
-              slackSend(channel: "${env.SLACK_CHANNEL}", color: 'RED', message: "${env.SERVICE}: Building and pushing docker image has failed - ${env.BUILD_URL}")
+              currentBuild.description = env.DOCKER_TAG
             }
           }
         }
         stage('Deploy to staging') {
           steps {
-            // this must match the "RUBY VERSION" in https://github.com/socrata/apps-marathon/blob/main/Gemfile.lock
-            withRbenv("2.5.3") {
-              script {
-                marathon.checkoutAndInstall()
-                marathon.deploy(env.SERVICE, 'staging', env.DOCKER_TAG)
-              }
+            script {
+              lastStage = env.STAGE_NAME
+              // Deploys env.SERVICE with tag env.DOCKER_TAG to staging by default
+              marathonDeploy()
             }
           }
-          post {
-            success {
-              slackSend(channel: "${env.SLACK_CHANNEL}", color: 'GREEN', message: "${env.SERVICE} (${env.DOCKER_TAG}): Successfully deployed to staging marathon")
-            }
-            failure {
-              slackSend(channel: "${env.SLACK_CHANNEL}", color: 'RED', message: "${env.SERVICE} (${env.DOCKER_TAG}): Failed deploying to staging marathon - ${env.BUILD_URL}")
-            }
+        }
+      }
+      post {
+        failure {
+          script {
+            teamsMessage(message: "Build [${currentBuild.fullDisplayName}](${env.BUILD_URL}) has failed in stage ${lastStage}", webhookCredentialID: WEBHOOK_ID)
```

Comment:
Note that this post is on the 'Build' stage not the 'PR' stage so PR failures will not be reported.